core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -579,6 +596,19 @@ private object RTS {
                     val io = curIo.asInstanceOf[IO.Ensuring[E, Any]]
                     stack.push(new Finalizer(io.finalizer))
                     curIo = io.io
+
+                  case IO.Tags.RunSync =>
+                    val io = curIo.asInstanceOf[IO.RunSync[E, Any]]
+                    stack.push(io)
+                    curIo = io.io
+
+                  case IO.Tags.FiberDefects =>
```

Comment:
Would there be any value in also checking this under the left-hand side for the `FlatMap` case? Also the checks for `map` and `bimap`.

Reply:
I would guess no, because it's always nested, not a one-step op like Point/Sync/Fail that can be optimized on flatMap.

Reply:
A, for FiberDefects, yes, it can be optimized. But it's supposedly a very rare op, so this may not be necessary.